Sources/Sentry/SentryNetworkTracker.m

```diff
@@ -265,6 +290,99 @@ - (void)urlSessionTask:(NSURLSessionTask *)sessionTask setState:(NSURLSessionTas
     SENTRY_LOG_DEBUG(@"SentryNetworkTracker finished HTTP span for sessionTask");
 }
 
+- (void)captureEvent:(NSURLSessionTask *)sessionTask
+{
+    NSHTTPURLResponse *myResponse = (NSHTTPURLResponse *)sessionTask.response;
+    NSNumber *responseStatusCode = [NSNumber numberWithLongLong:myResponse.statusCode];
+
+    if (!self.isCaptureFailedRequests || ![self containsStatusCode:myResponse.statusCode]) {
+        return;
+    }
+
+    if (![self isTargetMatch:sessionTask.currentRequest.URL
+                 withTargets:SentrySDK.options.failedRequestTargets]) {
+        return;
+    }
+    
+    NSString *message = [NSString
+        stringWithFormat:@"HTTP Client Error with status code: %ld", (long)myResponse.statusCode];
+
+    SentryEvent *event = [[SentryEvent alloc] initWithLevel:kSentryLevelError];
+
+    SentryThreadInspector *threadInspector = SentrySDK.currentHub.getClient.threadInspector;
+    // TODO: getCurrentThreads does not return stack traces
+    NSArray<SentryThread *> *threads = [threadInspector getCurrentThreadsWithStackTrace];
+
+    SentryException *sentryException = [[SentryException alloc] initWithValue:message
+                                                                         type:@"HTTP-ClientError"];
+    sentryException.mechanism =
+        [[SentryMechanism alloc] initWithType:@"SentryNetworkTrackingIntegration"];
+
+    if (threads.count > 0) {
+        SentryStacktrace *sentryStacktrace = [threads[0] stacktrace];
+        sentryStacktrace.snapshot = @(YES);
+
+        sentryException.stacktrace = sentryStacktrace;
+        // TODO: do I need this?
+        //    [threads enumerateObjectsUsingBlock:^(SentryThread *_Nonnull obj, NSUInteger idx,
+        //        BOOL *_Nonnull stop) { obj.current = [NSNumber numberWithBool:idx == 0]; }];
```

Comment:
@brustolin do I need to do this? the ANR integration does.

Reply:
No. ANR runs in its own thread, so we fell like changing the current thread for the main thread, that is the one being monitored. I dont think you should do this here.